listeners/measured.go

```diff
@@ -46,11 +96,23 @@ func (c *wrapMeasuredConn) OnState(s http.ConnState) {
 // Responds to the "measured" message type
 func (c *wrapMeasuredConn) ControlMessage(msgType string, data interface{}) {
 	if msgType == "measured" {
-		c.Conn.ID = data.(string)
+		ctxUpdate := data.(map[string]interface{})
+		c.ctxMx.Lock()
+		defer c.ctxMx.Unlock()
+		newContext := make(map[string]interface{}, len(c.ctx))
+		// Copy context
+		for key, value := range c.ctx {
+			newContext[key] = value
+		}
+		// Update context
+		for key, value := range ctxUpdate {
+			newContext[key] = value
+		}
+		c.ctx = newContext
```

Comment:
Hey I think we could be more fine-grained with the locking here right @oxtoacart? As in:

```
c.ctxMx.Lock()
c.ctx = newContext
c.ctxMx.Unlock()
```